src/main/java/org/gridsuite/filter/server/FilterService.java

```diff
@@ -91,10 +92,13 @@ public FilterService(FiltersToGroovyScript filtersToScript,
             .collect(Collectors.toList());
     }
 
-    List<IFilterAttributes> getFilters(List<UUID> ids) {
+    List<FilterAttributes> getFiltersMetadata(List<UUID> ids) {
         return filterRepositories.entrySet().stream()
-            .flatMap(entry -> entry.getValue().getFiltersAttributes(ids))
-            .collect(Collectors.toList());
+                .flatMap(entry -> entry.getValue().getFiltersAttributes(ids)).map(filter -> {
+                    filter.setType(filter.getType().equals(FilterType.SCRIPT) ? FilterType.SCRIPT : FilterType.FILTER);
```

Comment:
Add a comment here to explain why you are doing this. It could be strange for another dev to read this code later.